### ISSUE TITLE
[BUG] DataLakeFileSystemClient.GetDeletedPaths() not return all values in responds body

### ISSUE THREAD

**Issue by blueww** [Azure/azure-sdk-for-net#37649]
### Library name and version

Azure.Storage.Files.DataLake 12.14.0

### Describe the bug

By run DataLakeFileSystemClient.GetDeletedPaths(), The returned type [PathDeletedItem](https://github.com/Azure/azure-sdk-for-net/blob/4ad5d513a7d21c21a11228569021d31da0da76be/sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathDeletedItem.cs#L11) is very simple, only contains properties Path/DeletionId/DeletedOn/RemainingRetentionDays.

But the responds from server contains much more information (e.g. Content-Length), following is one of the item returned in the responds body.
```
</Blob><Blob><Name>dir1/dir2/dir4/file</Name><DeletionId>133340361492907131</DeletionId><Deleted>true</Deleted><Properties><Creation-Time>Mon, 17 Jul 2023 02:55:38 GMT</Creation-Time><Last-Modified>Mon, 17 Jul 2023 02:55:38 GMT</Last-Modified><Expiry-Time>Tue, 18 Jul 2023 02:55:49 GMT</Expiry-Time><Etag>0x8DB86714D0AD089</Etag><ResourceType>file</ResourceType><Content-Length>1024</Content-Length><Content-Type>application/octet-stream</Content-Type><Content-Encoding /><Content-Language /><Content-CRC64 /><Content-MD5>kdOBlavCN6IxTt2D3rMAmA==</Content-MD5><Cache-Control /><Content-Disposition /><BlobType>BlockBlob</BlobType><AccessTier>Hot</AccessTier><AccessTierInferred>true</AccessTierInferred><ServerEncrypted>true</ServerEncrypted><DeletedTime>Mon, 17 Jul 2023 02:55:49 GMT</DeletedTime><RemainingRetentionDays>0</RemainingRetentionDays></Properties><OrMetadata /></Blob>
```

Some Powershell customer try to calculate the deleted ADLSGen2 items occupied space. But current .net SDK seems not support it.

### Expected behavior

SDK should show the properties returned from server when call DataLakeFileSystemClient.GetDeletedPaths().

### Actual behavior

SDK doesn't show the properties returned from server when call DataLakeFileSystemClient.GetDeletedPaths().

### Reproduction Steps

1. Run DataLakeFileSystemClient.GetDeletedPaths()
2. In network trace, in the responds body , each returned items has many properties 
3. In SDK returned object, only Path/DeletionId/DeletedOn/RemainingRetentionDays be returned.

### Environment

dotnet --info
.NET SDK:
 Version:   7.0.306
 Commit:    f500069cb7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.306\

Host:
  Version:      7.0.9
  Architecture: x64
  Commit:       8e9a17b221

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  6.0.120 [C:\Program Files\dotnet\sdk]
  6.0.315 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]
  7.0.306 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by amnguye** [comment]
Yeah we can look into making sure it parses out all that information that we're missing in the `PathDeletedItem`. Looks like we may have to make some updates to the generated code here as well.
